core/src/main/java/org/frankframework/stream/Message.java

```diff
@@ -865,7 +861,16 @@ public static byte[] asByteArray(Object object, String defaultCharset) throws IO
 		if (object instanceof byte[]) {
 			return (byte[]) object;
 		}
-		return Message.asMessage(object).asByteArray(defaultCharset);
+		if (object instanceof Message) {
+			return ((Message) object).asByteArray(defaultCharset);
+		}
+		if (object instanceof MessageWrapper) {
+			return ((MessageWrapper<?>) object).getMessage().asByteArray(defaultCharset);
+		}
+		// In other cases, message can be closed directly after converting to byte array.
+		try (Message message = Message.asMessage(object)) {
+			return message.asByteArray(defaultCharset);
+		}
```

Comment:
I think a case should be added for `object instanceof String`, String already has a charset embedded and shouldn't use the `defaultCharset` parameter.

Reply:
DefaultCharset is not used in the current new code. Do we still need to add the `instanceof String`, do you think?

Reply:
I think that we should because a `byte[]` can be created from a `String` directly without having a `Message` inbetween.

And we should really get rid of abusing `Message` for this sort of casting of random data to random other data... (Niels created an issue for that, this morning).

Reply:
Yes, I agree. Added this one as well, for the `Message.asByteArray(Object object)` method.